### DISCUSSION TITLE
How can I make a table in Orchid Laravel Package with data from array of objects?

### DISCUSSION THREAD

**Discussion by Linux-Alex** [orchidsoftware/platform#2496] [Q&A]
I'm making an WooCommerce administration center with the [Laravel WooCommerce Package](https://codexshaper.github.io/docs/laravel-woocommerce/) and [Orchid - Laravel Admin Package](https://orchid.software/).

I'm getting product data from Shops in an array format: [Example](https://i.stack.imgur.com/2g3RS.png)

And I need to display this product in a table like this: [Example](https://i.stack.imgur.com/zIyf5.png)

For this I need to make an WooCommerceProductListLayout that extends from the class Table. But this extended class can get data only from a database. I don't want all the data from variables saving to a database, because the data is very large and temporary.

I made the WooCommerceProductListLayout class, but I don't know how to avoid getting data from database. All the data should come from Product::all() or WooCommerce::all('products') and not the database.

They use static method from:

```php
use Codexshaper\WooCommerce\Facades\Product;
use Codexshaper\WooCommerce\Facades\WooCommerce;
```

Please help me, thx

**Comment by tabuna** [comment]
Hi @Linux-Alex You need to wrap each element as a `Repository`. For example:

```php
use Orchid\Screen\Repository;

public function query(): iterable
{
    $productRawFromQuery = []; // Your array

    $products = collect($productRawFromQuery)->map(fn(array $items) => new Repository($items));

    return [
        'products' => $products,
    ];
}

public function layout(): iterable
{
    return [
        Layout::table('products', [
            TD::make('name'),
        ]),
    ];
}
```
